src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,40 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][i];
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int j = matrix.length - 1;
+        for (int i = 0; i < matrix.length; i++) {
+            counterDiagonal[i] = matrix[i][j];
+            j--;
+        }
+        return counterDiagonal;
     }
 
     public int getMin(int[][] matrix) {
-        return 0;
+        int minElement = matrix[0][0];
+        for (int i = 0; i < matrix.length; i++) {
+            for (int j = 0; j < matrix.length; j++) {
+                minElement = matrix[i][j] < minElement ? matrix[i][j] : minElement;
```

Comment:
in this case simple  if condition would look better, Agree?

Reply:
but Math.min() implementation looks like this. Therefore, I thought it was better.
I get it now.